runtimes/common/src/bonded_currencies/mod.rs

```diff
@@ -0,0 +1,291 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::{
+	dispatch::DispatchResult,
+	traits::{
+		fungible::{self, UnionOf},
+		fungibles::{self, metadata::Inspect},
+		tokens::{
+			AssetId as AssetIdTraits, DepositConsequence, Fortitude, Precision, Preservation, Provenance,
+			WithdrawConsequence,
+		},
+		AccountTouch, Get,
+	},
+};
+use sp_runtime::{
+	traits::Convert,
+	DispatchError, Either,
+	Either::{Left, Right},
+};
+use sp_std::{marker::PhantomData, vec::Vec};
+use substrate_fixed::types::{I75F53, U75F53};
+
+pub mod hooks;
+
+/// The AssetId for bonded assets.
+pub type AssetId = u32;
+
+/// Fixed point number used for creating bonding curves.
+pub type FloatInput = U75F53;
+
+/// Fixed point number used for doing calculation steps in the bonding curves.
+pub type Float = I75F53;
```

Comment:
why do we call these floats even though they aren't?

Reply:
Do you have a better name in mind? `FixedPoint` maybe?

Reply:
Yeah, that works

Reply:
Updated in:[0ff6f03](https://github.com/KILTprotocol/kilt-node/pull/818/commits/0ff6f03673c168f11a300f0f141b51824213083f)